py_src/ipyelk/diagram/elk_text_sizer.py

```diff
@@ -0,0 +1,68 @@
+"""Simple widget to get text size from DOM
+"""
+import asyncio
+import logging
+
+# Copyright (c) 2020 Dane Freeman.
+# Distributed under the terms of the Modified BSD License.
+import traitlets as T
+from ipywidgets import DOMWidget
+
+from .._version import EXTENSION_NAME, EXTENSION_SPEC_VERSION
+
+logger = logging.getLogger()
+
+
+class ElkTextSizer(DOMWidget):
+    """Jupyterlab widget for getting rendered text sizes from the DOM"""
+
+    _model_name = T.Unicode("ELKTextSizerModel").tag(sync=True)
+    _model_module = T.Unicode(EXTENSION_NAME).tag(sync=True)
+    _model_module_version = T.Unicode(EXTENSION_SPEC_VERSION).tag(sync=True)
+    _view_name = T.Unicode("ELKTextSizerView").tag(sync=True)
+    _view_module = T.Unicode(EXTENSION_NAME).tag(sync=True)
+    _view_module_version = T.Unicode(EXTENSION_SPEC_VERSION).tag(sync=True)
+
+    text = T.Unicode().tag(sync=True)
+    width = T.Float().tag(sync=True)
+    height = T.Float().tag(sync=True)
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.futures = {}
+        self.on_msg(self._resolve_measurement)
+        self.queue = asyncio.Queue()
```

Comment:
signal private as `_queue`?

Reply:
Yes definitely need that private. Also the line above is the line that isn't working at the moment...  I am not getting the measurement message back at the self._resolve_measurement function.